core/src/main/scala/spinal/core/Trait.scala

```diff
@@ -810,6 +810,8 @@ object noLatchCheck                  extends SpinalTag
 object noBackendCombMerge            extends SpinalTag
 object crossClockDomain              extends SpinalTag{ override def moveToSyncNode = true }
 object crossClockBuffer              extends SpinalTag{ override def moveToSyncNode = true }
+case class crossClockFalsePath(source: Option[BaseType] = None, destIsReset: Boolean = false) extends SpinalTag { override def allowMultipleInstance: Boolean = false }
```

Comment:
Or should we have a sealed trait for the destination type instead?

Reply:
Ahhhh yes, that is right, like a enum, yes yes

Reply:
fixed

Reply:
if we build this with an enum and keep the current prohibition for multiple tags then we can' have a path were both data and clock get a false path constraint - I have to admit that I can't think of a use-case for that one ATM, but reset & clock does make some sense...
Not sure having separate booleans isn't the preferable solution here?

Reply:
So IMO it doesn't prevent a path having both data and reset/clock enable?  A path is defined by the start point to endpoint and this tag is added to the first buffer (i.e. destination) of the timing path, so we shouldn't have any conflicts, no?

Reply:
Currently the tag has `allowMultipleInstances` set to `false`. Combine that with the current enum approach - you can only have a single tag on a given path, meaning you can't e.g. have a false path on both clock and reset.
(since you'd have to place two tags in the same path, which `allowMultipleInstances` should not allow - and the constraint generator would also have to be able to deal with).

That's why I thought the approach with separate booleans was the better one in this case.

Reply:
I still don't understand.  When you say _clock_, do you mean the clock enable signal in a CD?  My understanding is that even if the start point is different, the end point is always distinct between CE and reset, no?

Reply:
Lets say I wanted to do have the false path applied to both reset & clock enable, like so:
```scala
def doSmthg(x: Bool) = {
  x.addTag(crossClockFalsePath(destType=TimingEndpointType.RESET)
  x.addTag(crossClockFalsePath(destType=TimingEndpointType.CLOCK_EN)
}
```
this would currently not work

Reply:
I checked with
```scala
case class Test() extends Component {
  val io = new Bundle {
    val i = in port Bool()
    val o = out port Bool()
  }
  io.o := BufferCC(io.i, inputAttributes = List(crossClockFalsePath(destType=TimingEndpointType.RESET), crossClockFalsePath(destType=TimingEndpointType.CLOCK_EN)))
}
```
which gives the error
```
[error]  Conflicting tags added to the same item! crossClockFalsePath(None,RESET) ; crossClockFalsePath(None,CLOCK_EN)
```

Reply:
Uh but what you're doing doesn't make sense: you're not actually using it to drive the clock enable/reset signal.  If you look at this on the netlist level, clock enable (CE) and reset (PRE, etc.) are always _separate_ pins and thus create separate paths; you can always add the attribute on the non-shared parts of the paths and not having to worry about tag conflict.

Reply:
I think the example is actually not that esoteric - Altera recommends a synchronizer using the CE input of a FF, depending on where the reset for that is coming from you need the constraint on both of them.

Looks like this: https://www.intel.com/content/www/us/en/programmable/quartushelp/current/index.htm#da_rules/cdc_50102.htm
Personally I think these should then be delay constraints instead of false path constraints, but I've seen the latter used for that.

Instead of allowing multiple instances I'd go for multiple arguments, much shorter, just as extendible and if used with explicit parameter names just as readable.

Reply:
@KireinaHoro
How should we continue here? Modify it or leave as it is and drop the uniqueness requirement once we really need it?

Reply:
I'm with leaving it as is.  We can always come back when such a use case comes up

Reply:
I'll merge this then, thanks!